app/controllers/users_controller.rb

```diff
@@ -1,4 +1,20 @@
 class UsersController < ApplicationController
+  def new
+    @user = User.new
+  end
+
+  def create
+    @user = User.new(user_params)
+
+    if @user.save
+      session[:user_id] = @user.id
+      flash[:notice] = "Successfully Registered. You are now logged in."
+      redirect_to root_path
+    else
+      render 'new'
```

Comment:
I think you need to set the error flash message here, like:
flash.now.alert = @user.errors.full_messages.join(', ')
or something like that.  Flash.now lets it show with a render, rather than a redirect.